openverifiablellm/utils.py

```
@@ -114,6 +114,68 @@ def compute_sha256(file_path: Union[str, Path]) -> str:
+    directory_path : Union[str, Path]
+        Path to directory to hash.
+
+    Returns
+    -------
+    str
+        SHA256 hex digest representing directory contents.
+    """
+    directory = Path(directory_path)
+
+    if not directory.exists() or not directory.is_dir():
+        raise FileNotFoundError(f"Directory not found: {directory}")
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)